src/main/java/org/shanerx/tradeshop/Updater.java

```diff
@@ -27,161 +27,159 @@
+                    log.log(Level.WARNING, "[Updater] Please notice that the build may contain critical bugs!");
+                    log.log(Level.WARNING, "[Updater] +-----------------------------------------------------+");
+                    in.close();
+                    return RelationalStatus.AHEAD;
+                } else {
+                    log.log(Level.WARNING, "[Updater] You are running the latest version of the plugin!");
+                    in.close();
+                    return RelationalStatus.UP_TO_DATE;
+                }
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+            log.log(Level.WARNING, "[Updater] +----------------------------------------------------+");
+            log.log(Level.WARNING, "[Updater] Could not establish a connection to check for updates!");
+            log.log(Level.WARNING, "[Updater] +----------------------------------------------------+");
+        }
+        return RelationalStatus.UNKNOWN;
+    }
+
+    public RelationalStatus compareVersions(final String major, final String minor, final String patch) {
+        try {
+            return compareVersions(Short.parseShort(major), Short.parseShort(minor), Short.parseShort(patch));
+        } catch (NumberFormatException e) {
+            throw new IllegalArgumentException("All arguments must be numbers!");
+        }
+    }
+
+    public RelationalStatus compareVersions(final short major, final short minor, final short patch) {
+        try {
+            short selfMajor = Short.parseShort(getVersionComponent(SemVer.MAJOR));
+            short selfMinor = Short.parseShort(getVersionComponent(SemVer.MINOR));
+            short selfPatch = Short.parseShort(getVersionComponent(SemVer.PATCH));
+
+            if (selfMajor == major && selfMinor == minor && selfPatch == patch) {
+                return RelationalStatus.UP_TO_DATE;
+            } else if (selfMajor > major) {
+                return RelationalStatus.AHEAD;
+            } else if (selfMajor < major) {
+                return RelationalStatus.BEHIND;
+            } else if (selfMinor > minor) {
+                return RelationalStatus.AHEAD;
+            } else if (selfMinor < minor) {
+                return RelationalStatus.BEHIND;
+            } else if (selfPatch > patch) {
+                return RelationalStatus.AHEAD;
+            } else {
+                return RelationalStatus.BEHIND;
+            }
+
+        } catch (NumberFormatException e) {
+            throw new IllegalArgumentException("All arguments must be numbers!");
+        }
+    }
+
+    public enum BuildType {
+
+        DEV,
+
+        BETA,
+
+        STABLE,
+
+        FINAL
+    }
+
+    public enum SemVer {
+
+        MAJOR,
+
+        MINOR,
+
+        PATCH,
+
+        BUILD_TYPE
+    }
+
+    public enum RelationalStatus {
+
+        AHEAD,
+
+        UP_TO_DATE,
+
+        BEHIND,
+
+        UNKNOWN
+    }
```

Comment:
Could have been in a separate commit, but fine.

Is this simply a cleanup or am I missing something obvious?

Reply:
I didnt change anything here but i have it reformat the code before i commit, so probably just that

Reply:
Makes sense. No objections to that.